pkg/endpoint/policy.go

```diff
@@ -689,9 +690,10 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	e.BuildMutex.Lock()
 	defer e.BuildMutex.Unlock()
 
-	e.Mutex.Lock()
-	e.getLogger().Debug("Regenerating endpoint...")
-	e.Mutex.Unlock()
+	e.Mutex.RLock()
+	scopedLog := e.getLogger()
+	e.Mutex.RUnlock()
+	scopedLog.Debug("Regenerating endpoint...")
```

Comment:
This locking is no longer needed as `e.getLogger()` takes care of it's own locking.

Reply:
getLogger still needs the Endpoint.Mutex RLocked, in order to create the logger object from the fields, I'm afraid. We could make a public GetLogger that does this, to avoid the uglier lock/unlock combos here.
If we do, we could also unwind the `InsertEvent` function, which is basically a wrapper for getLogger just so the endpointmanager can use the endpoint's logger (https://github.com/cilium/cilium/blob/e845685472c2000c893ab3005839df04c3dfa8f3/pkg/endpoint/endpoint.go#L2517-L2519)